lang/java/wiredtiger.i

```diff
@@ -1820,6 +1877,132 @@ WT_ASYNC_CALLBACK javaApiAsyncHandler = {javaAsyncHandler};
 			return new PackInputStream(valueFormat,
 			    get_value_wrap(), _java_raw());
 	}
+
+	/**
+	 * Modify an existing record.
+	 *
+	 * The cursor must already be positioned, and the key's value will be
+	 * updated.
+	 *
+	 * \param mods an array of modifications.
+	 * \return 0 on success, errno on error.
+	 */
+	public int modify(Modify mods[])
+	throws WiredTigerException {
+		WT_MODIFY_LIST l = new WT_MODIFY_LIST(mods.length);
+		int pos = 0;
+
+		for (Modify m : mods) {
+			l.set(pos, m);
+			pos++;
+		}
+		return (_modify(l));
+	}
+%}
+
+/*
+ * Support for WT_CURSOR.modify.
+ */
+
+%inline %{
+typedef struct __wt_java_item_hold {
+#ifndef SWIG
+	void *data;
+	size_t size;
+	JNIEnv *jnienv;
+	jobject ref;
+#endif
+} WT_ITEM_HOLD;
+
+/*
+ * An internal Python class encapsulates a list of Modify objects (stored as a
+ * WT_MODIFY array in C).
+ */
+typedef struct __wt_java_modify_list {
+#ifndef SWIG
+	WT_MODIFY *mod_array;
+	jobject *ref_array;
+	JNIEnv *jnienv;
+	int count;
+#endif
+} WT_MODIFY_LIST;
+%}
+%extend __wt_java_modify_list {
+	__wt_java_modify_list(int count) {
+		WT_MODIFY_LIST *self =
+		    (WT_MODIFY_LIST *)calloc(1, sizeof(WT_MODIFY_LIST));
+		self->mod_array = (WT_MODIFY *)calloc((size_t)count,
+		    sizeof(WT_MODIFY));
+		self->ref_array = (jobject *)calloc((size_t)count,
+		    sizeof(jobject));
+		self->count = count;
```

Comment:
Should we check the `calloc` calls for errors? And should we be calling `__wt_calloc_def`?